### DISCUSSION TITLE
Spell Check plugin

### DISCUSSION THREAD

**Discussion by tscahill** [zim-desktop-wiki/zim-desktop-wiki#2280] [Q&A]
This discussion topic is about the fairly constant challenge I've experienced getting the spell-check plugin to function properly. Through a succession of Linux distributions, (A list of just those distributions that were successfully, but temporarily, installed is [here.](https://github.com/zim-desktop-wiki/zim-desktop-wiki/files/10344977/Distributions.txt) ) I have discovered that one or both of the indicated dependencies have not been included in that distribution's default repositories. In some, Zim itself was missing, or wasn't a recent release. The dependency gtkspell has been largely unavailable across many different distributions. In addition, the presence of a dependency has not been a guarantee that the spell check will function correctly. In several cases, the effort was disrupted by a system crash and migration to another distro. 

At other times, such as just now, with MX Linux, the spell check wasn't working, and then it was. Unfortunately, I have not been documenting as I now would have prefer, but during an ongoing cycle of crash/reboot/change distro/repeat, I often had more pressing issues on my plate (i.e., taxes, birthdays, routines, etc).

In this latest case, I have been running Zim 0.75.1 from source, by invoking zim.py.  From there, Zim's plugin Spell Checker view was reporting:
• gtkspellcheck - OK
• gtkspell - _Failed_

Initially, spell check wasn't underlining spelling errors, so I restarted it. After that it was underlining the errors, but not providing suggestions. Today, I thought I'd explore MX Linux repositories for anything similar to gtkspell, and I installed 

- libgtkspell-dev (2.0.16-1.3)

About sixty dependencies ( [dependency_additions.txt](https://github.com/zim-desktop-wiki/zim-desktop-wiki/files/10344916/dependency_additions.txt) ) were installed along with that package,

Possibly relevant in this list were:
`libgtk2.0-dev (2.24.33-2) libgtkspell-dev (2.0.16-1.3) libgtkspell0 (2.0.16-1.3)`

After restarting Zim, the dependency OK/Failed report was unchanged, and the lack of suggestions initially continued. _A few minutes in, and the suggestions suddenly became available_. Spell check is now functioning correctly, but there has been no change in the dependency report.  For my own needs, I'd like to document what changed to get the spell check functioning. Obviously, I should start by trying to isolate which particular package it was that I installed that provided just the right thing to get functionality. This would involve removing, one-at-a-time, the additional packages that came along with libgtkspell-dev. In my experience, this sort of task should be labelled: **_OPEN CAN OF WORMS_**. 

I intend to fold Zim-related tasks and issues such as this on into a larger mid-term project to create a selection of task-specific live USB distributions sharing a common desktop. (For example, one of the task-specific live USBs will be centered upon learning and practising Python). During this, I hope to revisit the default status of Zim on each of the half-dozen distributions I'm going to settle upon. This will follow naturally, due to the centrality of Zim in my all around record keeping. 

I'm putting this out here now to invite any suggestions or thoughts specific to troubleshooting the spell-check plugin. 

Also, if there is any enthusiasm for establishing a base-line for the status of Zim across the various Linux distributions, I would be happy to keep folks updated in a different discussion.

**Comment by introt** [comment]
Related issue: #2260 - you don't need to have both, a "good" version of either `gtkspellcheck` or `gtkspell` is enough.

To simplify your quest and to have the widest reach I'd suggest you focus on getting spell check working on the [Flatpak version of Zim](https://github.com/flathub/org.zim_wiki.Zim) - this way you don't need to modify your base system to figure out the appropriate packages, and you'll only need to go through the process once.
